core/trino-main/src/main/java/io/trino/sql/planner/sanity/VerifyUseTableScanPartitioningSet.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.planner.sanity;
+
+import io.trino.Session;
+import io.trino.execution.warnings.WarningCollector;
+import io.trino.metadata.Metadata;
+import io.trino.spi.type.TypeOperators;
+import io.trino.sql.planner.TypeAnalyzer;
+import io.trino.sql.planner.TypeProvider;
+import io.trino.sql.planner.plan.PlanNode;
+import io.trino.sql.planner.plan.TableScanNode;
+
+import static io.trino.sql.planner.optimizations.PlanNodeSearcher.searchFrom;
+
+public final class VerifyUseTableScanPartitioningSet
+        implements PlanSanityChecker.Checker
+{
+    @Override
+    public void validate(PlanNode plan,
+            Session session,
+            Metadata metadata,
+            TypeOperators typeOperators,
+            TypeAnalyzer typeAnalyzer,
+            TypeProvider types,
+            WarningCollector warningCollector)
+    {
+        searchFrom(plan)
+                .where(TableScanNode.class::isInstance)
+                .<TableScanNode>findAll()
+                .stream()
+                .filter(scan -> scan.getUseConnectorTablePartitioning().isEmpty())
+                .forEach(ignored -> {
```

Comment:
Maybe use `findFirst` to get a node not matching the required condition and also add details about the node in the exception